initfiles/sbin/hpcc_setenv.in

```diff
@@ -23,30 +23,46 @@
 ###<REPLACE>###
 
 function kill_process () {
-    SENTINEL=$1
-    PID=$2
-    TIMEOUT=$3
-    if [ -e $SENTINEL ]; then
-        rm -f $SENTINEL
+    local PID=$1
+    local PROCESS=$2
+    local TIMEOUT=$3
+    if [[ $# -eq 4 ]]; then
+      SENTINEL=$4
+      [[ -e $SENTINEL ]] && rm -f $SENTINEL
     fi
-    if [ -e $PID ]; then
-        pidwait_fn $PID $TIMEOUT
+    if [[ -e $PID ]]; then
+        pidwait_fn $PID $PROCESS $TIMEOUT
     fi
 }
 
 function pidwait_fn () {
-    PID=$1
-    WATCH_PID=`cat $PID`
-    TIMEOUT=$(($2*1000))
+    local PID=$1
+    local WATCH_PID=$( cat $PID )
+    local PROCESS=$2
+    local TIMEOUT=$(($3*1000))
+
+    if ps -p $WATCH_PID -o command= | grep -v "${PROCESS}" &>/dev/null ; then
+      exit 1
+    fi
+
     kill $WATCH_PID
-    while [ -d /proc/$WATCH_PID -a $TIMEOUT -gt 0 ]; do
-            sleep 0.2
-            TIMEOUT=$(($TIMEOUT-200))
+    while [[ -d /proc/$WATCH_PID ]] && [[ $TIMEOUT -gt 0 ]]; do
+      sleep 0.2
+      local TIMEOUT=$(($TIMEOUT-200))
     done
-    if [ $TIMEOUT -le 0 ]; then
-            kill -9 $WATCH_PID
+
+    if [[ $TIMEOUT -le 0 ]] && ps -p $WATCH_PID -o command= | grep "${PROCESS}" &>/dev/null ; then
+      kill -9 $WATCH_PID &>/dev/null
+      sleep 1
+      kill -0 $WATCH_PID &>/dev/null
+      # if still alive
+      if [[ $? -eq 0 ]]; then
+        return 1
```

Comment:
Does anything check return result of pidwait_fn() ?

Didn't it used to do something different here, i.e. have detected zombie.. 
Am not really sure what best thing to do is here.. if process won't die.., probably just log it.
It might be more pertinent to detect and abort out on start if there's an existing process around..